From: Roel Kluin <roel.kluin@gmail.com>
Date: Tue, 10 Aug 2010 18:01:10 -0700
Subject: [PATCH] gdth: unmap ccb_phys when scsi_add_host() fails in gdth_eisa_probe_one()
X-Git: 2c076eea6d3005c54f6e7be5938477fdc7027686
References: FATE#311692
Patch-Mainline: 2.6.34

unmap ccb_phys as well when scsi_add_host() fails

Signed-off-by: Roel Kluin <roel.kluin@gmail.com>
Cc: Achim Leubner <achim_leubner@adaptec.com>
Cc: James E.J. Bottomley <James.Bottomley@HansenPartnership.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/gdth.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/scsi/gdth.c b/drivers/scsi/gdth.c
index 246cedc..775ef54 100644
--- a/drivers/scsi/gdth.c
+++ b/drivers/scsi/gdth.c
@@ -4911,7 +4911,7 @@ static int __devinit gdth_eisa_probe_one(u16 eisa_slot)
 
 	error = scsi_add_host(shp, NULL);
 	if (error)
-		goto out_free_coal_stat;
+		goto out_free_ccb_phys;
 	list_add_tail(&ha->list, &gdth_instances);
 	gdth_timer_init();
 
-- 
1.6.0.2

